shared/src/test/scala/squants/market/MoneySpec.scala

```diff
@@ -152,17 +152,39 @@ class MoneySpec extends FlatSpec with Matchers {
     y.moneyMin(x) should be(y)
   }
 
-  it should "return proper result when adding like currencies with no MoneyContext in scope" in {
+  it should "return a proper result when adding like currencies with no MoneyContext in scope" in {
     USD(1) + USD(2) should be(USD(3))
     USD(1).plus(USD(2)) should be(USD(3))
   }
 
-  it should "return proper result when subtracting like currencies with no MoneyContext in scope" in {
+  it should "return a proper result when subtracting like currencies with no MoneyContext in scope" in {
     USD(5) - USD(2) should be(USD(3))
     USD(5).minus(USD(2)) should be(USD(3))
   }
 
-  it should "return proper result when dividing like currencies with no MoneyContext in scope" in {
+  it should "return a proper result with no additional precision loss when multiplying by a primitive numeric" in {
```

Comment:
So, the Scalachecks revealed that this issue can also exist with BigDecimal.

```scala
val x = BigDecimal(3)
val m = BigDecimal(2)

m / x + m / x = 1.3333333333333333333333333333333334
(m / x) * 2 = 1.333333333333333333333333333333333
```

Interestingly, that same calculation with Double works correctly

```scala
val x = BigDecimal(3)
val m = BigDecimal(2)

m / x + m / x = 1.3333333333333333
(m / x) * 2 = 1.3333333333333333
```

I can resolve this by using an approximation with a tolerance of USD(1e-32) to the comparison.  Most uses cases will use much less than 32 digits and will typically round to the nearest penny (or formattedDecimal for the Currency), so this is still a big improvement over the 16/17 digits we get with Double math.

The multiplication use case noted in the original issue does not have this problem, so no approximation was needed there.

I'll have an updated PR in a moment.